### DISCUSSION TITLE
Kubeflow Pipelines 2.3.x planning

### DISCUSSION THREAD

**Discussion by rimolive** [kubeflow/pipelines#10908] [General]
Hello, Pipelines community!

The Pipelines WG is planning the future releases of KFP and we'd like to hear from you what are the features you want to see in the coming releases.

Let's use this thread to discuss the most desired features, and don't forget to join Pipelines WG to discuss the roadmap for the next releases.

**Comment by tom-pavz** [comment]
Hello! Here are a few issues I would love to see get into a sooner release than later for kfp!

1. https://github.com/kubeflow/pipelines/issues/10036 (total dealbreaker for using KFP v2 IMO)
2. https://github.com/kubeflow/kubeflow/issues/7422 (this issue seems to be have been created in the incorrect GH repo, but I am seeing it in my current 2.1.0 deploy of KFP, so I think it is valid)
3. https://github.com/kubeflow/pipelines/issues/10917
4. https://github.com/kubeflow/pipelines/issues/10241
5. https://github.com/kubeflow/pipelines/issues/10897

**Comment by postera-james** [comment]
Agreed! in particular https://github.com/kubeflow/pipelines/issues/10036 is an essential feature to replicate before a move to v2 is possible for anyone using any kind of scaling

**Comment by HumairAK** [comment]
Some more: 

* Documentation for s3-prefixes (the feature itself is already merged and will go out on the next 2.3 release) 
  * Relevant issue: https://github.com/kubeflow/pipelines/issues/9689
* https://github.com/kubeflow/pipelines/pull/10942  (implementation is almost complete)
* https://github.com/kubeflow/pipelines/issues/10977
  * if we can do argo 3.5 latest that's even better
* https://github.com/kubeflow/pipelines/issues/10187
* This is a priority in our downstream, and the work on this is almost done, we can forward this implementation to upstream kfp for 2.3: https://github.com/kubeflow/pipelines/issues/10872 
* https://github.com/kubeflow/pipelines/issues/10957
* https://github.com/kubeflow/pipelines/issues/8718 (implementation is basically done)
* https://github.com/kubeflow/pipelines/issues/10839
* https://github.com/kubeflow/pipelines/issues/10656

**Comment by StefanoFioravanzo** [comment]
We also need an issue to track V1/V2 feature parity. It's not necessarily related to 2.3.X, but more broadly we need a high-level understanding and summary of what's missing

**Comment by HumairAK** [comment]
With 2.3 out, it seems we didn't get quite all of these, so those issues unaccounted for should probably be pushed to the 2.4.x planning doc, or milestone if we get one created.

**Comment by rimolive** [comment]
Agreed. Closing this discussion, but let's work on creating the milestones for 2.4
